HabbitLogger.antonderegt/HabbitLogger/Program.cs

```diff
@@ -0,0 +1,254 @@
+﻿using Microsoft.Data.Sqlite;
+
+namespace HabbitLogger;
+
+class Program
+{
+    private readonly string connectionString = @"Data Source=habitLogger.db";
+
+    static void Main(string[] args)
+    {
+        bool keepRunning = true;
+        Program program = new();

```

Comment:
Did you initialized the class inside itself on purpose?
You can directly call methods and fields if you make them static.

```
class Program
{
    static private readonly string connectionString = @"Data Source=habitLogger.db";

    static void Main(string[] args)
    {
        bool keepRunning = true;
        
       EnsureTableExists();
       SeedData();

        while (keepRunning)
        {
            ShowMenu();
            keepRunning = ProcessInput();
        }
    }

    static void EnsureTableExists()
    {
        using ......
```

Reply:
Thanks, now you mention it, it does make more sense to keep everything static and pass the connectionString around. Also makes the methods more flexible.

Reply:
@antonderegt Although it's ok for these beginner projects I'll point out the overusing static isn't good practice as it defeats the purpose of object oriented programming. Essentially it should only be used with classes that don't contain state. Not to mention concurrency issues. 😊